### ISSUE TITLE
Problem with headless 

### ISSUE THREAD

**Issue by alexander-miglis** [AdrianStrugala/AvroConvert#45]
Hi,

I'm looking at your library, which seems really nice btw. But I could not get passed this simple example.

I'm using the AttributeClass in your example in the documentation here.

```csharp
  var schema = AvroConvert.GenerateSchema(typeof(AttributeClass));
  AttributeClass ac = new AttributeClass { 
      AndAnotherString = "asdf",
      AnotherWayOrIndicatingNullableProperty = 1, 
      NullableIntPropertyWithDefaultValue = null,
      StringProperty = "aaaaaaaa"
  };
			
  byte[] data = AvroConvert.SerializeHeadless(ac, schema);
  AttributeClass ac2 = AvroConvert.DeserializeHeadless<AttributeClass>(data, schema);
```
  
  When I run this the SerializeHeadless returns a 4 byte empty array, and offcourse the deserialize will only give me a default object. If I remove the decorations in AttributeClass, it will start working. It actually seems to be working for any property without decoration.
  
Am I doing something wrong here? Would very much appreciate how this can be solved.
Thanks

**Comment by AdrianStrugala** [comment]
Hello,

Thank you for creating the Issue. It looks like a bug at the first glance. I will investigate the problem and proceed with a fix.

Thanks,
Adrian

**Comment by AdrianStrugala** [comment]
Hello,

Your implementation is correct.
The problem is resolved and you can find the fix in version 3.1.1 of the AvroConvert.

Thanks,
Adrian

**Comment by alexander-miglis** [comment]
> Hello,
> 
> Your implementation is correct.
> The problem is resolved and you can find the fix in version 3.1.1 of the AvroConvert.
> 
> Thanks,
> Adrian

Wow, thank you for the quick response
